detectors/azure/vm/vm.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package vm // import "go.opentelemetry.io/contrib/detectors/azure/vm"
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.24.0"
+)
+
+const (
+	defaultAzureVMMetadataEndpoint = "http://169.254.169.254/metadata/instance/compute?api-version=2021-12-13&format=json"
+)
+
+type config struct {
+	endpoint string
+}
+
+func newConfig(options ...Option) config {
+	c := config{defaultAzureVMMetadataEndpoint}
+	for _, option := range options {
+		c = option.apply(c)
+	}
+
+	return c
+}
+
+// Option applies an Azure VM detector configuration option.
+type Option interface {
+	apply(config) config
+}
+
+type optionFunc func(config) config
+
+func (fn optionFunc) apply(c config) config {
+	return fn(c)
+}
+
+// WithEndpoint sets the endpoint for obtaining a Azure instance metadata JSON.
+func WithEndpoint(e string) Option {
```

Comment:
In what cases would someone need to set this?  Or is this just for testing?

Do we still want to keep our parameters for version and format?

Reply:
This is for testing, I don't currently see another use case.

Reply:
Lets not add a public function for `WithEndpoint` then.  You can still keep `Option` around for future extensibility. For testing, you can just set the endpoint field on the struct directly.

Reply:
Done, I removed the options altogether.